module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp

```diff
@@ -131,66 +233,108 @@ namespace module::localisation {
             // Use configuration here from file RobotParticleLocalisation.yaml
-            filter.model.processNoiseDiagonal = config["process_noise_diagonal"].as<Expression>();
-            filter.model.n_rogues             = config["n_rogues"].as<int>();
-            filter.model.resetRange           = config["reset_range"].as<Expression>();
-            filter.model.n_particles          = config["n_particles"].as<int>();
-            draw_particles                    = config["draw_particles"].as<int>();
-
-            Eigen::Vector3d start_state = config["start_state"].as<Expression>();
-            // TODO: This variable is not used. Probably remove it
-            /* Eigen::Vector3d start_variance = config["start_variance"].as<Expression>(); */
-
-            auto reset = std::make_unique<ResetRobotHypotheses>();
-            ResetRobotHypotheses::Self leftSide;
-            // Start on goal line
-            leftSide.position     = Eigen::Vector2d(start_state.x(), start_state.y());
-            leftSide.position_cov = Eigen::Vector2d::Constant(0.5).asDiagonal();
-            leftSide.heading      = start_state.z();
-            leftSide.heading_var  = 0.005;
-
-            reset->hypotheses.push_back(leftSide);
-            ResetRobotHypotheses::Self rightSide;
-            // Start on goal line
-            rightSide.position     = Eigen::Vector2d(start_state.x(), -start_state.y());
-            rightSide.position_cov = Eigen::Vector2d::Constant(0.5).asDiagonal();
-            rightSide.heading      = -start_state.z();
-            rightSide.heading_var  = 0.005;
-
-            reset->hypotheses.push_back(rightSide);
-            emit<Scope::DELAY>(reset, std::chrono::seconds(1));
+            filter.model.processNoiseDiagonal = cfg["process_noise_diagonal"].as<Expression>();
+            filter.model.n_rogues             = cfg["n_rogues"].as<int>();
+            filter.model.resetRange           = cfg["reset_range"].as<Expression>();
+            filter.model.n_particles          = cfg["n_particles"].as<int>();
+
+            config.debug = cfg["debug"].as<bool>();
+
+            config.start_variance = cfg["start_variance"].as<Expression>();
+        });
+
+        on<Startup, With<FieldDescription>>().then([this](const FieldDescription& fd) {
+            // Left side penalty mark
+            config.start_state.emplace_back((-fd.dimensions.field_length / 2.0) + fd.dimensions.penalty_mark_distance,
+                                            (-fd.dimensions.field_width / 2.0),
+                                            -M_PI_2);
+
+            // Right side penalty mark
+            config.start_state.emplace_back((-fd.dimensions.field_length / 2.0) + fd.dimensions.penalty_mark_distance,
+                                            (fd.dimensions.field_width / 2.0),
+                                            M_PI_2);
+
+            // // Left side goal line
+            // config.start_state.emplace_back(
```

Comment:
commented

Reply:
We should be using these starting poses in the filter, however it's unstable atm with these here

Reply:
@Bidski Thoughts? Also apparently we can spawn in one of three places on each side of the field currently, so we should add those in @ysims told me. For now I think this is fine

Reply:
In what places can we spawn? Obviously is makes the filter unstable so we shouldn't do it.

If there are multiple places along one side of the field we could have that as a single start point with a large variance in one axis?